### PR TITLE
Use the upstream kvdb

### PR DESCRIPTION
**Pull request by sgkim126**
There are two major changes.
- kvdb-rocksdb removed WAL support
- kvdb::Error is replaced with io::Error

### CODE REVIEW

**Comment by Remagpie** [comment]
[kvdb 0.1.1](https://crates.io/crates/kvdb) still uses `elastic-array` crate.
I'm a bit concerned about it considering that we merged https://github.com/CodeChain-io/codechain/pull/1870.

**Comment by sgkim126** [comment]
@Remagpie It's not a concern, because kvdb uses ElasticArray internally and doesn't expose it.

**Comment by Remagpie** [comment]
[`DBValue`](https://docs.rs/kvdb/0.1.1/kvdb/type.DBValue.html) is exposed, and is an alias of `ElasticArray128<u8>`.
I think we will be using this type a lot.

**Comment by sgkim126** [comment]
@Remagpie I missed it.
However, I think it's better to keep track of the upstream.

File: rpc/src/v1/impls/devel.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ where
     }
 
     fn get_state_trie_value(&self, key: H256) -> Result<Vec<Bytes>> {
-        match self.db.get(COL_STATE, &key).map_err(|e| errors::kvdb(&e))? {
+        match self.db.get(COL_STATE, &key).map_err(errors::transaction_core)? {
```
Comment (Remagpie) [inline rpc/src/v1/impls/devel.rs]: Is this correct?

File: core/src/client/error.rs [hunk 2]
```diff
@@ -16,14 +16,12 @@
 
 use std::fmt::{Display, Error as FmtError, Formatter};
 
-use kvdb;
 use util_error::UtilError;
 
 /// Client configuration errors.
 #[derive(Debug)]
 pub enum Error {
     /// Database error
```
Comment (Remagpie) [inline core/src/client/error.rs]: Please remove this comment too.

File: rpc/src/v1/impls/devel.rs (hunk 1 above)
Comment (sgkim126) [reply rpc/src/v1/impls/devel.rs]: Yes, because there is no more kvdb error.
The error becomes io::Error.

Comment (Remagpie) [reply rpc/src/v1/impls/devel.rs]: I think it'll be interpreted as `unknown_error` in `transaction_core`, which has a message `"Unknown error when sending transaction"`.
I think using `errors::core` is better here.

Comment (sgkim126) [reply rpc/src/v1/impls/devel.rs]: Looks good. I changed.

**Comment by Remagpie** [comment]
@sgkim126 Is it still work in progress?

**Comment by sgkim126** [comment]
@Remagpie It's done. I'll merge it when Travis passes.

**Pull request merged**